sv_pipeline/load_data.py

```diff
@@ -0,0 +1,392 @@
+
+
+def get_es_schema(all_fields, nested_fields):
+    schema = {
+        key: {'type': ES_FIELD_TYPLE_MAP.get(key) or ES_TYPE_MAP[type(val[0]) if isinstance(val, list) else type(val)]}
+        for key, val in all_fields.items() if key not in nested_fields
+    }
+    for key, val_dict in nested_fields.items():
+        schema[key] = {'type': 'nested', 'properties': get_es_schema(val_dict, {})}
+    return schema
+
+
+def get_es_index_name(project, meta):
+    return '{project}__structural_variants__{sample_type}__grch{genome_version}__{datestamp}'.format(
+        project=project,
+        sample_type=meta['sampleType'],
+        genome_version=meta['genomeVersion'],
+        datestamp=datetime.today().strftime('%Y%m%d'),
+    ).lower()
+
+
+def export_to_elasticsearch(es_host, es_port, rows, index_name, meta):
+    es_client = ElasticsearchClient(host=es_host, port=es_port)
+
+    all_fields = {}
+    nested_fields = {GENOTYPES_FIELD: {}, TRANSCRIPTS_FIELD: {}}
+    for row in rows:
+        all_fields.update({k: v for k, v in row.items() if v})
+        for col, val in nested_fields.items():
+            val.update(row[col][0])
+    elasticsearch_schema = get_es_schema(all_fields, nested_fields)
+
+    if es_client.es.indices.exists(index=index_name):
+        print('Deleting existing index')
+        es_client.es.indices.delete(index=index_name)
+
+    print('Setting up index')
+    es_client.create_or_update_mapping(
+        index_name, ES_INDEX_TYPE, elasticsearch_schema, num_shards=NUM_SHARDS, _meta=meta
+    )
+
+    es_client.route_index_to_temp_es_cluster(index_name, True)
+
+    es_actions = [{
+        '_index': index_name,
+        '_type': ES_INDEX_TYPE,
+        '_op_type': ELASTICSEARCH_INDEX,
+        '_id': row[VARIANT_ID_FIELD],
+        '_source': row,
+    } for row in rows]
+
+    print('Starting bulk export')
+    success_count, _ = es_helpers.bulk(es_client.es, es_actions, chunk_size=1000)
+    print('Successfully created {} records'.format(success_count))
+
+    es_client.es.indices.forcemerge(index=index_name)
+
+    es_client.route_index_to_temp_es_cluster(index_name, False)
+
+
+if __name__ == '__main__':
+    p = argparse.ArgumentParser()
+    p.add_argument('input_dataset', help='input VCF or VDS')
+    p.add_argument('--skip-sample-subset', action='store_true')
+    p.add_argument('--write-subsetted-bed', action='store_true')
+    p.add_argument('--ignore-missing-samples', action='store_true')
+    p.add_argument('--in-silico')
+    p.add_argument('--project-guid')
+    p.add_argument('--es-host', default='localhost')
+    p.add_argument('--es-port', default='9200')
+
+    args = p.parse_args()
+
+    sample_subset = None
+    sample_remap = None
+    if not args.skip_sample_subset:
+        sample_subset = get_sample_subset(args.project_guid)
+        sample_remap = get_sample_remap(args.project_guid)
+        message = 'Subsetting to {} samples'.format(len(sample_subset))
+        if sample_remap:
```

Comment:
this assumes the remap file will always exist for projects that need remapping, right? though I'm not sure how we would provide a list of projects that need remapping. I guess that would need to be a manual check

Reply:
nope! `get_sample_remap` just returns `None` if the file doesn't exist. A good argument for why docstrings are helpful